pkg/restore/restore_test.go

```diff
@@ -2113,6 +2112,100 @@ func TestRestorePersistentVolumes(t *testing.T) {
 	}
 }
 
+type fakeResticRestorerFactory struct {
+	podVolumeBackups []*velerov1api.PodVolumeBackup
+}
+
+func (f *fakeResticRestorerFactory) NewRestorer(ctx.Context, *velerov1api.Restore) (restic.Restorer, error) {
+	return &fakeResticRestorer{
+		podVolumeBackups: f.podVolumeBackups,
+	}, nil
+}
+
+type fakeResticRestorer struct {
+	podVolumeBackups []*velerov1api.PodVolumeBackup
+}
+
+func (b *fakeResticRestorer) RestorePodVolumes(data restic.RestoreData) []error {
+	return nil
+}
+
+// TestBackupWithRestic runs backups of pods that are annotated for restic backup,
+// and ensures that the restic backupper is called, that the returned PodVolumeBackups
+// are added to the Request object, and that when PVCs are backed up with restic, the
+// claimed PVs are not also snapshotted using a VolumeSnapshotter.
+func TestRestoreWithRestic(t *testing.T) {
+	var (
+		backupName = "backup-1"
+		veleroNs   = "velero"
+	)
+
+	tests := []struct {
+		name         string
+		restore      *velerov1api.Restore
+		backup       *velerov1api.Backup
+		apiResources []*test.APIResource
+		tarball      io.Reader
+		want         []*velerov1api.PodVolumeBackup
+	}{
+		{
```

Comment:
You'll want to set up test cases that have:
- PodVolumeBackups associated with the backup that are being passed in via the Request obj
- Pods in the tarball that the PodVolumeBackups are linked to

Then you'll want to verify that `RestorePodVolumes` is called as expected. TBH this could be a place where using a mock would make sense, since we can't really verify much more than that the method was called as expected (# times, args specified), given the nature of how restic works and that we're not actually running in a cluster.